### PR TITLE
dockerfile-language-server 0.11.0 (new formula)

### PR DESCRIPTION
**Pull request by gwynne**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Add formula for `dockerfile-language-server-nodejs`.

### CODE REVIEW

**Comment by chenrui333** [comment]
- raised an question about the repo name, https://github.com/rcjsuen/dockerfile-language-server-nodejs/issues/268

**Comment by chenrui333** [comment]
the commit msg should be either `dockerfile-language-server 0.11.0 (new formula) ` or `dockerfile-language-server-nodejs 0.11.0 (new formula)` (checking with upstream about the repo name, I personally prefer `dockerfile-language-server` for the formula name)

**Comment by gwynne** [comment]
100% agreed regarding the name. I'll wait a day or two before updating the commit message to see if the upstream responds.

**Comment by chenrui333** [comment]
Looks like upstream does not accept/reject the idea, let's move forward with the `dockerfile-language-server` (just for the sake of consistency of all the LSP formulae)

**Comment by gwynne** [comment]
Sounds good to me 👍

**Comment by rcjsuen** [comment]
Hey all, thank you for your interest in my project. Could we put a hold on this for the time being assuming this isn't time-sensitive?

Full disclosure I do not work for Docker Inc and am in no ways associated with Docker and/or the Moby Project as an external third-party contributor either. I don't want to step on any toes regarding the "officialness" of my hobby project (due to suggestive naming conventions per se) and have reached out to a contact at Docker Inc about this. Apologies for this just want to try to run it by Docker Inc first.

**Comment by chenrui333** [comment]
@rcjsuen thanks for the context and help moving this thing forward!

**Comment by rcjsuen** [comment]
> As dockerfile is actually an open source standard, I don't think this is not an licensing/trademark issue. Just my two cents.

Yeah, I know what you mean. @chenrui333 I don't know anything about Homebrew's internals but how hard is it to rename a package after the fact? It feels like it would inconvenience your users, correct?

**Comment by chenrui333** [comment]
> > As dockerfile is actually an open source standard, I don't think this is not an licensing/trademark issue. Just my two cents.
> 
> Yeah, I know what you mean. @chenrui333 I don't know anything about Homebrew's internals but how hard is it to rename a package after the fact? It feels like it would inconvenience your users, correct?

we can do `formula_renames.json` to help with renaming thing, but if we can avoid from the beginning, why not doing from the start :)

**Comment by rcjsuen** [comment]
Hi all, thanks for accommodating and your patience on this matter. I have gotten a verbal okay to proceed and have renamed my GitHub repository accordingly.

Let me know if you need me to do anything else to help move this along and thank you @gwynne for helping get my project into Homebrew!

**Comment by gwynne** [comment]
It's my pleasure! And thank you in turn for your work on the project itself, it's coming very much in handy for me in combination with BBEdit already ❤️

**Comment by gwynne** [comment]
@rcjsuen Belated follow-up: Will the package name in NPM change as well?

File: Formula/d/dockerfile-language-server.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+require "language/node"
+
+class DockerfileLanguageServer < Formula
+  desc "Language server for Dockerfiles powered by Node, TypeScript, and VSCode"
+  homepage "https://github.com/rcjsuen/dockerfile-language-server-nodejs"
```
Comment (chenrui333) [inline Formula/d/dockerfile-language-server.rb:5]: the repo can be renamed cc @gwynne

**Comment by chenrui333** [comment]
> Hi all, thanks for accommodating and your patience on this matter. I have gotten a verbal okay to proceed and have renamed my GitHub repository accordingly.
> 
> Let me know if you need me to do anything else to help move this along and thank you @gwynne for helping get my project into Homebrew!

that is lovely, thanks!!

File: Formula/d/dockerfile-language-server.rb (hunk 1 above)
Comment (gwynne) [reply Formula/d/dockerfile-language-server.rb]: Done - I'd actually done it already, but I fat-fingered when I force-pushed the squash and didn't notice until now 😅

**Comment by rcjsuen** [comment]
> @rcjsuen Belated follow-up: Will the package name in NPM change as well? (I have no particular opinion on whether or not it should, just need to know for the formula to be correct 😅)

That's less of a simple change since that would require deprecating the package and so on. I don't have any immediate plans to take action there at the moment.

**Comment by gwynne** [comment]
> That's less of a simple change since that would require deprecating the package and so on. I don't have any immediate plans to take action there at the moment.

👍 Makes sense! (Although you should probably fix the repo and homepage links on [the npmjs.com page](https://www.npmjs.com/package/dockerfile-language-server-nodejs), and also the various badges in the README. Sorry to be such a stickler for detail; I'm infamously a nitpicker, just ask anyone who has to work with me! 😅)

So in that case, as far as I can tell, the formula is ready to go! /cc @chenrui333

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @gwynne @rcjsuen for the awesome work! 🎉

**Comment by rcjsuen** [comment]
> 👍 Makes sense! Although you should probably fix the repo and homepage links on [the npmjs.com page](https://www.npmjs.com/package/dockerfile-language-server-nodejs), and also the various badges in the README. (Sorry to be such a stickler for detail; I'm infamously a nitpicker, just ask anyone who has to work with me! 😅)

No need to apologize, you are absolutely right about that. GitHub does offer redirects so those old links do work but it is better to correct them for sure.

**Pull request merged**